game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -424,37 +425,19 @@ private static void sortAaHighToLow(
     sortAaHighToLow(units, data, defending, new HashMap<>());
   }
 
-  private static void sortAaHighToLow(
+  @VisibleForTesting
+  static void sortAaHighToLow(
       final List<Unit> units,
       final GameData data,
       final boolean defending,
       final Map<Unit, Tuple<Integer, Integer>> unitPowerAndRollsMap) {
-    final Comparator<Unit> comparator =
-        (u1, u2) -> {
-          final Tuple<Integer, Integer> tuple1 =
-              getMaxAaAttackAndDiceSides(
-                  Collections.singleton(u1), data, defending, unitPowerAndRollsMap);
-          final Tuple<Integer, Integer> tuple2 =
-              getMaxAaAttackAndDiceSides(
-                  Collections.singleton(u2), data, defending, unitPowerAndRollsMap);
-          if (tuple1.getFirst() == 0) {
-            if (tuple2.getFirst() == 0) {
-              return 0;
-            }
-            return 1;
-          } else if (tuple2.getFirst() == 0) {
-            return -1;
-          }
-          final float value1 = ((float) tuple1.getFirst()) / ((float) tuple1.getSecond());
-          final float value2 = ((float) tuple2.getFirst()) / ((float) tuple2.getSecond());
-          if (value1 < value2) {
-            return 1;
-          } else if (value1 > value2) {
-            return -1;
-          }
-          return 0;
-        };
-    units.sort(comparator);
+    units.sort(
+        Comparator.comparing(
+            unit ->
+                getMaxAaAttackAndDiceSides(
+                    Collections.singleton(unit), data, defending, unitPowerAndRollsMap),
+            Comparator.<Tuple<Integer, Integer>, Boolean>comparing(tuple -> tuple.getFirst() == 0)
```

Comment:
A comment to describe the compare algo could be nice to have as well. Maybe extracting to a method would work too so the method name can be used to describe the operation that is occurring.

Reply:
I mean if I understood the algorithm that would be a feasible thing to do, but I don't unfortunately.